test/jdk/java/awt/FileDialog/MultipleMode.java

```diff
@@ -0,0 +1,105 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Button;
+import java.awt.Checkbox;
+import java.awt.FileDialog;
+import java.awt.FlowLayout;
+import java.awt.Frame;
+import java.awt.Panel;
+import java.awt.TextArea;
+import java.io.File;
+
+/*
+ * @test
+ * @bug 6467204
+ * @summary Need to implement "extended" native FileDialog for JFileChooser
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual MultipleMode
+*/
+
+public class MultipleMode {
+
+    private static final String INSTRUCTIONS =
+            """
+             1. Turn the 'multiple' checkbox off and press the 'open' button
+             2. Verify that the file dialog doesn't allow the multiple file selection
+             3. Select any file and close the file dialog
+             4. The results will be displayed, verify the results
+             5. Turn the 'multiple' checkbox on and press the 'open' button
+             6. Verify that the file dialog allows the multiple file selection
+             7. Select several files and close the file dialog
+             8. The results will be displayed, verify the results.
+            """;
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame
+            .builder()
+            .title("MultipleMode test instructions")
+            .instructions(INSTRUCTIONS)
+            .rows(15)
+            .columns(40)
+            .testUI(MultipleMode::init)
+            .build()
+            .awaitAndCheck();
+    }
+
+    private static Frame init() {
+        Frame frame = new Frame("MultipleMode");
+        TextArea sysout = new TextArea("", 20, 80);
+        sysout.setEditable(false);
+
+        final Checkbox mode = new Checkbox("multiple", true);
+
+        Button open = new Button("open");
+        open.addActionListener(e -> {
+            FileDialog d = new FileDialog(frame);
+            d.setMultipleMode(mode.getState());
+            d.setVisible(true);
+
+            // print the results
+            sysout.append("DIR:\n");
+            sysout.append("  %s\n".formatted(d.getDirectory()));
+            sysout.append("FILE:\n");
+            sysout.append("  %s\n".formatted(d.getFile()));
+            sysout.append("FILES:\n");
+            for (File f : d.getFiles()) {
+                sysout.append("  %s\n".formatted(f));
+            }
+        });
+
+        Panel panel = new Panel(new FlowLayout());
+        panel.add(mode);
+        panel.add(open);
+
+        frame.setLayout(new BorderLayout());
+        frame.add(panel, BorderLayout.NORTH);
+        frame.add(sysout, BorderLayout.CENTER);
+
+        frame.pack();
```

Comment:
Frame size is too big and going out of the screen, might fix a size here.

Reply:
It doesn't fit on the screen of my VM either.

One way to resolve this problem is to add
```java
            .position(PassFailJFrame.Position.TOP_LEFT_CORNER)
```
to `PassFailJFrame` builder as there's no easy way to move the instruction frame to the left so that both frames fit on the screen.

Reply:
Updated.

What resolution and scale are you using?
I've tested at 1920x1080 and default scale 1, and it has plenty of room: 
![image](https://github.com/openjdk/jdk/assets/77687766/5ab8035b-7d50-402b-a26d-0cbade786673)

Reply:
Now its fine on my screen. But its not in center, but top. Does it differ for windows and linux?

Reply:
It is not in the center, because in the latest commit I added `.position(PassFailJFrame.Position.TOP_LEFT_CORNER)`, as Alexey suggested.

Reply:
I have 1920×1200 with 150%. This is why it doesn't fit. The second frame overlapped instructions a bit.

Reply:
That would be 1280x800 user pixels ?

These days tests should be able to assume that there are 1920x1080 user-space pixels on the screen.